docs/message-schema-definitions.md

```diff
@@ -0,0 +1,166 @@
+# Message Schema Definitions and Samples
+
+## Cosmos DB Collections
+
+### On Hand
+
+This collection stores one document for each product in every store. It is partitioned by the `id` property, which is composed of `divId:storeId:upc`. A change feed listens to this collection and will get documents that have been changed or updated in batches. The schema of these incoming documents matches the schema of the original defined below.
+
+Schema: [On Hand](./json/schemas/input-onhand-schema.json)
+
+Example:
+
+```json
+{
+    "id": "div1:store005:0003400029005",
+    "divisionId": "div1",
+    "storeId": "store005",
+    "upc": "0003400029005",
+    "inventoryCount": 1000,
+    "productName": "HERSHEY'S Milk Chocolate Bars",
+    "description": "HERSHEY'S Milk Chocolate Bars are the classic full size chocolate candy bars you’ve always enjoyed!",
+    "lastUpdateTimestamp": "2020-05-13 9:00:00 AM"
+}
+```
+
+Link: [On Hand Example](./json/samples/input-onhand-sample.json)
+
+### Shipments
+
+This collection stores one document for each shipment delivered to any store. It is partitioned by the `id` property, which is a GUID representing the shipment id. A change feed listens to this collection and will get documents that have been changed or updated in batches. The schema of these incoming documents matches the schema of the original defined below. The change feed processor will send one request to the Orchestrator for each item in the items array for the shipment.
```

Comment:
so for N documents we have N partitions? should we have a better partition key? typically don't we desire for a partition to span a (somewhat equal-sized) group of items?

Reply:
Hmm looks like my comment didn't post.. sorry about that! 

For Cosmos DB `id` is usually a really good choice for partition key because it has a very high cardinality (1:1) and will evenly spread your data across all partitions. Since the workload for this collection is write heavy and doesn't really get queried, we just need to focus on spreading out our writes.